DFC.ServiceTaxonomy.GraphSync/Activities/AuditSyncIssuesTask.cs

```diff
@@ -0,0 +1,144 @@
+            return Outcomes(T["Done"]);
+#pragma warning restore S3220 // Method calls should not resolve ambiguously to overloads with "params"
+        }
+
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var timestamp = DateTime.UtcNow;
+            var log = (await _session.Query<AuditSyncLog>().FirstOrDefaultAsync()) ??
+                      new AuditSyncLog() {LastSynced = DateTime.MinValue};
+
+            var contentItems = await _session
+                //do we only care about the latest published items?
+                .Query<ContentItem, ContentItemIndex>(x =>
+                    x.ContentType.IsIn(_contentTypes.Keys) && x.Latest && x.Published && (x.CreatedUtc >= log.LastSynced ||  x.ModifiedUtc >= log.LastSynced))
+                .ListAsync();
+
+            foreach (var contentItem in contentItems)
+            {
+                await CheckIfContentItemSynced(contentItem);
+            }
+
+            foreach (var contentItem in _syncFailedContentItems)
+            {
+                try
+                {
+                    await _mergeGraphSyncer.SyncToGraph(contentItem.ContentType, contentItem.Content,
+                        contentItem.CreatedUtc, contentItem.ModifiedUtc);
+
+                    _syncFailedContentItems.Remove(contentItem);
+
+                    await CheckIfContentItemSynced(contentItem);
+                }
+                catch
+                {
+                    //I don't want to do anything here, why won't this build without this comment?
+                }
+            }
+
+            //anything left in the syncFailedContentItems collection is still failing to sync - report this somehow
+
+            log.LastSynced = timestamp;
+            _session.Save(log);
+            await _session.CommitAsync();
+
+            return Outcomes("Done");
+        }
+
+        private async Task CheckIfContentItemSynced(ContentItem contentItem)
+        {
+            var results = await _graphDatabase.RunReadQuery(new MatchNodeWithAllOutgoingRelationshipsQuery(contentItem.ContentType, contentItem.Content.GraphSyncPart.Text));
+
+            if (results == null || !results.Any())
+            {
+                _syncFailedContentItems.Add(contentItem);
+            }
+            else
+            {
+                var contentDefinition = _contentTypes[contentItem.ContentType];
+
+                var sourceNode = results.Select(x => x[0]).Cast<INode>().First();
+                var relationships = results.Select(x => x[1]).Cast<IRelationship>().ToList();
+                var destNodes = results.Select(x => x[2]).Cast<INode>().ToList();
+
+                foreach (var part in contentDefinition.Parts)
+                {
+                    if (!_partSyncers.TryGetValue(part.PartDefinition.Name, out var partSyncer))
```

Comment:
We should only use the eponymous graph syncer if it's definitely the eponymous part (i.e. either set the _partSyncers key to the name of the content type, or try getting "Eponymous" if the part name is the same as the content type name).
By using the eponymous part syncer for all parts without a syncer, it'll have undefined behaviour if the content type contains a part there isn't a graph syncer for.